tests/phpunit/payment-methods/class-wc-stripe-upe-payment-gateway-test.php

```
@@ -4372,6 +4372,56 @@ public function test_add_email_currency_conversion_notice_outputs_plain_text_for
+	 * @see https://github.com/woocommerce/woocommerce-gateway-stripe/issues/4875
+	 *
+	 * @runInSeparateProcess
+	 * @preserveGlobalState disabled
+	 */
+	public function test_set_cookie_on_current_request_does_not_set_cookie_outside_checkout() {
+		// WOOCOMMERCE_CHECKOUT is not defined — simulates My Account registration context.
+		$gateway = new WC_Stripe_UPE_Payment_Gateway();
+		$gateway->set_cookie_on_current_request( 'test_cookie_value' );
+
+		$this->assertArrayNotHasKey( LOGGED_IN_COOKIE, $_COOKIE );
+	}
```

PHPMD (2.15.0)

[error] test_set_cookie_on_current_request_does_not_set_cookie_outside_checkout accesses the super-global variable $_COOKIE. (undefined)

(Superglobals)